src/redis-cli.c

```diff
@@ -418,91 +425,226 @@ static sds cliVersion(void) {
     return version;
 }
 
+// Concatenate a string to an sds string, but if it's empty substitute double quote marks.
+static sds sdscat_orempty(sds params, char *value) {
+    if (value[0] == '\0') {
+        return sdscat(params, "\"\"");
+    }
+    return sdscat(params, value);
+}
+
+static sds cliAddArgument(sds params, redisReply *argMap);
+
+// Concatenate a list of arguments to the parameter string, separated by a separator string.
+static sds cliConcatArguments(sds params, redisReply *arguments, char *separator) {
+    for (size_t j = 0; j < arguments->elements; j++) {
+        params = cliAddArgument(params, arguments->element[j]);
+        if (j != arguments->elements - 1) {
+            params = sdscat(params, separator);
+        }
+    }
+    return params;
+}
+
+// Add an argument to the parameter string.
+static sds cliAddArgument(sds params, redisReply *argMap) {
+    char *name = NULL;
+    char *type = NULL;
+    int optional = 0;
+    int multiple = 0;
+    int multipleToken = 0;
+    redisReply *arguments = NULL;
+    sds tokenPart = sdsempty();
+    sds repeatPart = sdsempty();
+
+    // First read the fields describing the argument.
+    for (size_t i = 0; i < argMap->elements; i += 2) {
+        char *key = argMap->element[i]->str;
+        if (!strcmp(key, "name")) {
+            name = argMap->element[i + 1]->str;
+        }
+        else if (!strcmp(key, "token")) {
+            char *token = argMap->element[i + 1]->str;
+            tokenPart = sdscat_orempty(tokenPart, token);
+        }
+        else if (!strcmp(key, "type")) {
+            type = argMap->element[i + 1]->str;
+        }
+        else if (!strcmp(key, "arguments")) {
+            arguments = argMap->element[i + 1];
+        }
+        else if (!strcmp(key, "flags")) {
+            redisReply *flags = argMap->element[i + 1];
+            for (size_t j = 0; j < flags->elements; j++) {
+                char *flag = flags->element[j]->str;
+                if (!strcmp(flag, "optional")) {
+                    optional = 1;
+                }
+                else if (!strcmp(flag, "multiple")) {
+                    multiple = 1;
+                }
+                else if (!strcmp(flag, "multiple_token")) {
+                    multipleToken = 1;
+                }
+            }
+        }
+    }
+
+    // Then build the "repeating part" of the argument string.
+    if (!strcmp(type, "key") ||
+        !strcmp(type, "string") ||
+        !strcmp(type, "integer") ||
+        !strcmp(type, "double") ||
+        !strcmp(type, "pattern") ||
+        !strcmp(type, "unix-time") ||
+        !strcmp(type, "token")) {
+            repeatPart = sdscat_orempty(repeatPart, name);
+    }
+    else if (!strcmp(type, "oneof")) {
+        repeatPart = cliConcatArguments(repeatPart, arguments, "|");
+    }
+    else if (!strcmp(type, "block")) {
```

Comment:
styling in redis is usually that on multi-line `if` the opening bracket is on the next line, but on `if-else` it's on the same line.
e.g. 
```suggestion
        !strcmp(type, "token"))
    {
        repeatPart = sdscat_orempty(repeatPart, name);
    } else if (!strcmp(type, "oneof")) {
        repeatPart = cliConcatArguments(repeatPart, arguments, "|");
    } else if (!strcmp(type, "block")) {
```

Reply:
Is there any kind of style guide?

Reply:
no.. just looking at the code around you and try to mimic it's style.
but sadly, redis-cli is not very consistent with the rest of redis (which isn't very consistent with itself)